src/lwm2m_coap_server_content.erl

```diff
@@ -22,23 +31,37 @@
 coap_discover(_Prefix, _Args) ->
     [].
 
-coap_get(_ChId, _Prefix, [], Query, _State) ->
+coap_get(_ChId, _Prefix, Query, _Content, State) ->
     Links = coap_core_link:encode(filter(lwm2m_coap_server_registry:get_links(), Query)),
-    #coap_content{etag = binary:part(crypto:hash(sha, Links), {0,4}),
-                  content_format = <<"application/link-format">>,
-                  payload = list_to_binary(Links)};
-coap_get(_ChId, _Prefix, _Else, _Query, _State) ->
-    {error, not_found}.
-
-coap_post(_ChId, _Prefix, _Suffix, _Content, _State) -> {error, method_not_allowed}.
-coap_put(_ChId, _Prefix, _Suffix, _Content, _State) -> {error, method_not_allowed}.
-coap_delete(_ChId, _Prefix, _Suffix, _State) -> {error, method_not_allowed}.
-
-coap_observe(_ChId, _Prefix, _Suffix, _Ack, _State) -> {error, method_not_allowed}.
-coap_unobserve(_State) -> ok.
-handle_info(_Message, State) -> {noreply, State}.
+    {ok, #coap_content{etag = binary:part(crypto:hash(sha, Links), {0,4}),
+                       content_format = <<"application/link-format">>,
+                       payload = list_to_binary(Links)}, State};
+
+coap_get(_ChId, _Prefix, _Query, _Content, State) ->
+    {error, not_found, State}.
```

Comment:
the return value change is also something not seemingly hot-upgradable unless the old code was crashing.

Reply:
or the caller supports both 2 and 3 element tuples ?

Reply:
As mentioned above